Gems/Atom/RHI/Code/Source/RHI/CpuProfilerImpl.cpp

```diff
@@ -173,6 +173,21 @@ namespace AZ
             return m_enabled;
         }
 
+        const CachedTimeRegion::GroupRegionName& CpuProfilerImpl::InsertDynamicName(const char* groupName, const AZStd::string& regionName)
+        {
+            AZStd::scoped_lock lock(m_dynamicNameMutex);
+            AZ_Warning("CpuProfiler", m_dynamicRegionNameMap.size() < 16384, "Stored dynamic region names are accumulating. Consider removing a AZ_ATOM_PROFILE_DYNAMIC invocation.");
+            auto [regionNameItr, inserted] = m_dynamicRegionNameMap.try_emplace(regionName, regionName);
+
+            // This is the first time we have constructed a dynamic region with this name, also create a GroupRegionName for the region
+            if (inserted)
+            {
+                m_dynamicGroupRegionNameMap.emplace(regionName, CachedTimeRegion::GroupRegionName{ groupName, regionNameItr->second.c_str() } );
+            }
+
+            return m_dynamicGroupRegionNameMap.at(regionName);
```

Comment:
This code doesn't work with this use case

InsertDynamicName("group1", "region1");
InsertDynamicName("group2", "region1"); // it returns group1_region1 since region1 is already exist in m_dynamicGroupRegionNameMap

Reply:
Good catch - I switched to using a nested map structure like in `ImGuiCpuProfiler` to prevent this collision between regions with different group names but the same region name. Thanks!